### PR TITLE
Make error handling for bad class work better

### PR DESCRIPTION
**Pull request by andy-faunadb**

### CODE REVIEW

File: src/model/Model.js [hunk 1]
```diff
@@ -268,7 +268,7 @@ export default class Model {
     if (!(typeof resource === 'object' && resource.constructor === Object))
       throw new Error('Expected to initialize from plain object resource.')
     const expectedClass = this.constructor.classRef
-    if (!resource.class.equals(expectedClass))
+    if (!(resource.class instanceof Ref && resource.class.equals(expectedClass)))
```
Comment (eaceaser) [inline src/model/Model.js]: i'd rewrite this as:

`!(resource.class instanceof Ref) || !resource.class.equals(expectedClass)`

Having to demorgan the negation makes it less clear what the intended logic is.

Comment (andy-faunadb) [inline src/model/Model.js]: This is why we need `unless`

Comment (eaceaser) [inline src/model/Model.js]: yuck! even worse, lol

**Comment by eaceaser** [comment]
lgtm

**Pull request merged**